ipa-core/src/report.rs

```diff
@@ -631,4 +634,173 @@ mod test {
+            ]
+            .reconstruct(),
+            expected.matchkey
+        );
+        assert_eq!(
+            [
+                dec_report1.breakdown_key,
+                dec_report2.breakdown_key,
+                dec_report3.breakdown_key
+            ]
+            .reconstruct(),
+            expected.breakdown_key
+        );
+        assert_eq!(
+            [
+                dec_report1.trigger_value,
+                dec_report2.trigger_value,
+                dec_report3.trigger_value
+            ]
+            .reconstruct(),
+            expected.trigger_value
+        );
+        assert_eq!(
+            [
+                dec_report1.timestamp,
+                dec_report2.timestamp,
+                dec_report3.timestamp
+            ]
+            .reconstruct(),
+            expected.timestamp
+        );
+    }
+
+    #[test]
+    fn check_compatibility_impressionmk_with_ios_encryption() {
+        let enc_report_bytes1 = hex::decode("12854879d86ef277cd70806a7f6bad269877adc95ee107380381caf15b841a7e995e414c63a9d82f834796cdd6c40529189fca82720714d24200d8a916a1e090b123f27eaf24f047f3930a77e5bcd33eeb823b73b0e9546c59d3d6e69383c74ae72b79645698fe1422f83886bd3cbca9fbb63f7019e2139191dd000000007777772e6d6574612e636f6d").unwrap();
+        let enc_report_bytes2 = hex::decode("1d85741b3edf3f49e8ed5824b8ea0ed156301fb6d450fc30ad76785fc3b281775937d0275efc237d3e3ac92e22cf60ebd8dc09a41abaa20c0a7ee9e5e1c736708c01dd65f592e5683f8ca0e23f8bfcd3a7736335cc5bec95beceb6474abb816b01f9adf7cc12c344c1538bb84c98b089b24733790032e70c7406000000007777772e6d6574612e636f6d").unwrap();
+        let enc_report_bytes3 = hex::decode("545f9df229a16c70497dd1f93ac75bef8ad33e836bb20f2ff37297bd814a091389d85db9007e7b95231a3e5a0055ae59dc56d431849c0aaf5e01e66c8e6b7888bf299f66907861798097aba96aae193d59b7fcafd5655e745f4b4ae51631c6342e36ee3b6f1682385b46295b7ce0128af02f6828cba562bf0c12000000007777772e6d6574612e636f6d").unwrap();
+
+        assert_eq!(enc_report_bytes1.len(), 138);
+        assert_eq!(enc_report_bytes2.len(), 138);
+        assert_eq!(enc_report_bytes3.len(), 138);
+
+        let expected = RawReport {
+            event_type: EventType::Source,
+            epoch: 0,
+            site_domain: String::from("www.meta.com"),
+            matchkey: 1,
+            trigger_value: 0,
+            breakdown_key: 45,
+            timestamp: 456,
+        };
+
+        validate_blobs(
+            &enc_report_bytes1,
+            &enc_report_bytes2,
+            &enc_report_bytes3,
+            &expected,
+        );
+    }
+
+    #[test]
+    fn check_compatibility_conversion_with_ios_encryption() {
+        let enc_report_bytes1 = hex::decode("741cd5012df1cf8f337258066a55c408d1052297af27a35bdef571773215ad7cbd367eab689145a24ad9666a12731a221ff5548cc7591a5ce50da4dcde203cc614175759ef230641adac977187143471b512f1c8fd95eafeb53602d90a69a6411f3af9cb44e02417f6f27b7162f08bff009e82b1c2c2aaaf156f010000007777772e6162632e636f6d").unwrap();
```

Comment:
If you end a line in a string literal with a backslash, rustc removes the backslash and any leading whitespace on the following line. This can be used to embed blobs like this while keeping the line length manageable.